ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -353,26 +362,23 @@ interested in this candidate at all.
 chainForkSuffix
   :: (HasHeader header, HasHeader block,
       HeaderHash header ~ HeaderHash block)
-  => ChainFragment block  -- ^ Current chain.
-  -> ChainFragment header -- ^ Candidate chain
-  -> Maybe (ChainFragment header)
-chainForkSuffix current candidate =
-    case ChainFragment.intersectChainFragments current candidate of
-      Nothing                         -> Nothing
-      Just (_, _, _, candidateSuffix) -> Just candidateSuffix
+  => AnchoredFragment block  -- ^ Current chain.
+  -> AnchoredFragment header -- ^ Candidate chain
+  -> AnchoredFragment header
+chainForkSuffix current candidate = candidateSuffix
+  where
+    (_, _, candidateSuffix) = AnchoredFragment.intersect current candidate
```

Comment:
Again, here's where any pre-condition or assumption on the anchors comes up.

Is this really total now? Do we guarantee that they're anchored at the same point, or that the candidate is anchored on the current chain? Is it totally impossible to not have an intersection?

Reply:
Fixed in the newest version.